sdk/cosmos/azure-cosmos/docs/Metrics.md

```diff
@@ -0,0 +1,174 @@
+| cosmos.client.req.rntbd.timeline.xxx                         | duration | 95th, 99th + histogram | Duration spent in different stages of the request pipeline   |
+| cosmos.client.req.reqPayloadSize                             |    bytes    | None                   | The request payload size in bytes                             |
+| cosmos.client.req.rspPayloadSize                             |    bytes    | None                   | The response payload size in bytes                             |
+| cosmos.client.req.rntbd.addressResolution.latency | duration | 95th, 99th + histogram | Duration spent for resolving physical addresses of replica for a certain partition |
+| cosmos.client.req.rntbd.stats.endpoint.acquiredChannels      | #        | 95th, 99th + histogram | Number of new TCP connections being established per Cosmos DB service endpoint |
+| cosmos.client.req.rntbd.stats.endpoint.availableChannels     | #        | None                   | Number of established TCP connections per  Cosmos DB service endpoint |
+| cosmos.client.req.rntbd.stats.endpoint.inflightRequests      | #        | 95th, 99th + histogram | Number of concurrently processed requests  per Cosmos DB service endpoint |
+
+### Metrics for RNTBD service endpoints (across operations, no operation-level tags)
+
+| Name                                            | Unit     | Percentiles            | Description                                                  |
+| ----------------------------------------------- | -------- | ---------------------- | ------------------------------------------------------------ |
+| cosmos.client.rntbd.requests.latency            | duration | 95th, 99th + histogram | End-to-end duration of processing RNTBD requests per service endpoint across all operations |
+| cosmos.client.rntbd.requests.failed.latency     | duration | 95th, 99th + histogram | End-to-end duration of failed RNTBD requests per service endpoint across all operations |
+| cosmos.client.rntbd.requests.successful.latency | duration | 95th, 99th + histogram | End-to-end duration of successful RNTBD requests per service endpoint across all operations |
+| cosmos.client.rntbd.req.reqSize                 | bytes    | 95th, 99th + histogram | Request payload size in RNTBD encoded bytes across requests for all operations per service endpoint |
+| cosmos.client.rntbd.req.rspSize                 | bytes    | 95th, 99th + histogram | Response payload size in RNTBD encoded bytes across requests for all operations per service endpoint |
+| cosmos.client.rntbd.endpoints.count             | snapshot | None                   | Current number (snapshot) of service endpoints with any active connections |
+| cosmos.client.rntbd.endpoints.evicted           | snapshot | None                   | Current number (snapshot) of endpoints for which TCP connections were closed |
```

Comment:
cosmos.client.rntbd.endpoints.count
cosmos.client.rntbd.endpoints.evicted 

I wonder whether we should hide these two metrics for now, the reason being: even if there is no active connections, SDK does not evicted the endpoint (immediately).

Reply:
These existed in rntbd metrics - I think it is fine to keep them- it is a snapshot (Gauge) anyway. Just want to make sure we don;t have to keep rntbdMetrics (the inital one) around for ever.

Reply:
And because it is just a gauge it is super cheap. Every interval just the getXX method is called and the current snapshot is reported in metrics. No min/max/avg, histogram or percentile.